.changelog/15808.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+core: enforce strict steps for clients reconnect
```

Comment:
This is a really important change, so probably worth being a little more explicit about it:

```suggestion
core: enforce strict ordering that node status updates are recorded after allocation updates for reconnecting clients
```